numpy/core/src/umath/ufunc_object.c

```diff
@@ -5893,13 +5893,55 @@ new_array_op(PyArrayObject *op_array, char *data)
     return (PyArrayObject *)r;
 }
 
+/*
+ * Use an indexed loop to do the work
+ * Returns 0 if successful
+ */
+static int
+trivial_at_loop(PyArrayMethodObject *ufuncimpl, NPY_ARRAYMETHOD_FLAGS flags,
+                    PyArrayMapIterObject *iter,
+                    PyArrayObject *op1_array, PyArrayObject *op2_array,
+                    PyArrayMethod_Context *context)
+{
+    int buffersize=0, errormask = 0;
+    int res;
+    char *args[3];
+    npy_intp dimensions = iter->size;
+    npy_intp steps[3];
+    args[0] = (char *) iter->baseoffset;
+    args[1] = (char *) iter->outer_ptrs[0];
+    args[2] = (char *)PyArray_DATA(op2_array);
+    steps[0] = iter->fancy_strides[0];
+    steps[1] = iter->outer_strides[0];
+    steps[2] = PyArray_STRIDE(op2_array, 0);
+    res = ufuncimpl->contiguous_indexed_loop(context, args, &dimensions, steps, NULL);
+    /*
+     * An error should only be possible if `res != 0` is already set.
+     * But this is not strictly correct since old-style ufuncs (e.g. `power`
+     * released the GIL but manually set an Exception).
+     */
+    if (PyErr_Occurred()) {
+        res = -1;
+    }
+
+    if (res == 0 && !(flags & NPY_METH_NO_FLOATINGPOINT_ERRORS)) {
```

Comment:
Is the matching FPE clearing before the actual inner-loop core missing here?

A bit hard to test for, I guess.  In practice doing a `np.inf / np.inf` in Python should cause an error already being set when entering the function, but I am not sure that it isn't cleared somewhere along the way (maybe that oddity cannot even be triggered).

Reply:
added the clear